src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -69,6 +71,9 @@
 public abstract class FlowNode extends Actionable implements Saveable {
     private transient List<FlowNode> parents;
     private final List<String> parentIds;
+    private transient List<FlowNode> enclosingBlocks;
+    private final List<String> allEnclosingIds = new ArrayList<>();
```

Comment:
🐛  We do not want to attach this to individual FlowNodes, and *definitely* don't want to add additional persistent content to them.  This is because I intend this information to be retained and exposed via a higher-level construct that provides fast search, probably tied to the FlowNodeStorage.  Per what I said earlier: 

> One note: we may want to ensure the actual implementation itself is never exposed in case (cough) this is supported directly via the FlowNodeStorage implementations in the future.

Once we have this tied into the storage (or a StorageIndex class) then having the extra persistent data on the nodes will be just cruft.

Reply:
Ok, I was confused by your earlier comments about persisting all the enclosing blocks here.

Reply:
Yeah we do not want a new non`transient` field here. (Especially one which is not restored by `readResolve` or some `onLoad` or whatever.)

Reply:
Yeah, waiting to see @svanoort's variant on this theme. =)

Reply:
It's coming.  I was working on it on the flight but couldn't get into the zone very well